html/entities.ts

```diff
@@ -115,6 +129,52 @@ export function unescape(
     .replaceAll(RX_HEX_ENTITY, (_, hex) => codePointStrToChar(hex, 16));
 }
 
+/**
+ * A valid custom element name is a sequence of characters name that meets all of the following requirements:
+ *
+ * name must match the PotentialCustomElementName production:
+ *
+ * PotentialCustomElementName ::= [a-z] (PCENChar)* '-' (PCENChar)*
+ *
+ * PCENChar ::= "-" | "." | [0-9] | "_" | [a-z] | #xB7 | [#xC0-#xD6] | [#xD8-#xF6] | [#xF8-#x37D] | [#x37F-#x1FFF] | [#x200C-#x200D] | [#x203F-#x2040] | [#x2070-#x218F] | [#x2C00-#x2FEF] | [#x3001-#xD7FF] | [#xF900-#xFDCF] | [#xFDF0-#xFFFD] | [#x10000-#xEFFFF]
+ *
+ * This uses the EBNF notation from the XML specification. [XML]
```

Comment:
These details may seem overwhelming to the reader. I think it's best that we just link to the relevant spec section for those that want to find out more.